spec/index.bs

```diff
@@ -1842,20 +1843,68 @@ interface GPUQueue {
     GPUFence createFence(optional GPUFenceDescriptor descriptor = {});
     void signal(GPUFence fence, unsigned long long signalValue);
 
+    void writeBuffer(
+        ArrayBuffer sourceData,
+        GPUBufferSize sourceOffset,
+        GPUBuffer destination,
+        GPUBufferSize destinationOffset,
+        GPUBufferSize size);
+
+    void writeTexture(
+        ArrayBuffer sourceData,
+        GPUTextureDataLayout sourceLayout,
```

Comment:
Does this have the same 256 limitation as for copies between buffers and textures?

Reply:
Good question!

I don't think it should have the limitation, since the data is coming from the CPU side, via a shortcut method, so we have both a room and a permit to manage the alignment automatically.

This could also be one more thing towards the list of why an API like this is useful. cc @jdashg

Reply:
I thought about this too, I think it should not have the limitation.

Reply:
If it doesn't have the limitation, it *must* copy bloat when this happens, which chips away at the "good enough performance" claims.

Reply:
I assume when @Kangz says "256 limitation" he's referring to buffer copies having to have strides that are a multiple of 256? (I ⌘f-ed for "256" in the spec and didn't find anything...)

If the source data just happens to have stride that is a multiple of 256, just by coincidence, then no copy bloat, right? Just like how, if `writeBuffer()` is called when the buffer just _happens_ to be ready, just by coincidence, then no copy bloat either, right? It doesn't make a lot of sense to mandate 256 here when we aren't mandating an internal copy.

Perhaps this can be alleviated with a new call which would be like "if I hypothetically ran a `writeBuffer()` with these arguments, would it be on the fast path or not?" and we can spec exactly what situations that call would return true or false (regardless of whether the underlying implementation has the 256 limitation or not).

Reply:
My comment wasn't super clear: the 256 limitation is around the rowPitchInBytes in `GPUTextureDataLayout`. `copyBufferToTexture` and `copyTextureToBuffer` require it to be a multiple of 256 and I was wondering if the same applied for `writeTexture`.

> Just like how, if writeBuffer() is called when the buffer just happens to be ready, just by coincidence, then no copy bloat either, right?

That's an implementation detail, and at least in Chromium/Dawn we'd like to avoid doing it if we can (it requires knowing for each resource when's the last time they were used by the GPU, which we'd like to avoid tracking).